server/src/serializers/ReviewSerializer.js

```diff
@@ -1,13 +1,17 @@
 class ReviewSerializer {
-  static async getReviewDetails(review) {
+  static async getReviewDetails(review, currentUser) {
     const allowedAttributes = ["id", "body", "planetId", "userId", "rating"]
     let serializedReview = {}
     for (const attribute of allowedAttributes) {
       serializedReview[attribute] = review[attribute]
     }
-
     serializedReview.voteCount = await review.$voteCount()
-  
+    serializedReview.currentUserVote = await review
+      .$relatedQuery("votes")
+      .findOne({ userId: currentUser.id, reviewId: review.id })
+    if (serializedReview.currentUserVote === undefined) {
+      serializedReview.currentUserVote = 0
+    }
```

Comment:
hmmm Im not sure this tracks. In your application, there is a distinction between a `currentUserVote` being 0 versus `null`. 0 means they have voted but unvoted. If a vote isnt present at all for this user, it should just be null unless this is causing major issues some place else.